### PR TITLE
cleanup common/models/base a bit

### PR DESCRIPTION
**Pull request by saneshark**
This begins to fix the issue with attributes not automatically updating in lib/common/models/base.rb

However, I'd also like to remove the automatic snakecasing of attributes that was previously taking place. I do not believe this is a responsibility for this model. It should happen prior to lib/common/models/base.rb being used, perhaps in a Faraday::Middleware similar to the one that Rx and Sm use to snakecase keys.

### CODE REVIEW

**Comment by saneshark** [comment]
@carlmjohnson, @austin-martinez 

disability_ratings.rb is going to fail specs after these changes ^ I believe it is because you guys were relying on lib/common/models/base.rb to snakecase keys, something I would like to remove.

**Comment by robbiethegeek** [comment]
@saneshark is this still being worked and by whom?

**Comment by carlmjohnson** [comment]
I don't follow how this affects snake casing. Isn't that handled by Olive Branch?

If this obviates the definition of `attributes` in `DisabilityClaimDocument`, that's a good thing, and we should remove it from there.

File: lib/common/models/base.rb [hunk 1]
```diff
@@ -5,12 +5,11 @@
...
-    include ActiveModel::Serialization
     extend ActiveModel::Naming
+    include ActiveModel::Serialization
+    include Comparable
     include Virtus.model(nullify_blank: true)
 
-    attr_reader :attributes
     attr_accessor :metadata, :errors_hash
     alias to_h attributes
     alias to_hash attributes
```
Comment (austin-martinez) [inline lib/common/models/base.rb]: I think you can remove these lines as well, they are defined in Virtus
https://github.com/solnic/virtus/blob/master/lib/virtus/instance_methods.rb#L45

**Comment by austin-martinez** [comment]
@saneshark  Could you remove L50-56 in `disability_claim_document.rb` as part of this PR (this is the use case that prompted this change) https://github.com/department-of-veterans-affairs/vets-api/blob/master/app/models/disability_claim_document.rb#L56

Also, you are correct about `disability_rating.rb`, the API is returning camelcased attributes but the old base would snakecase them, so we became dependent on that. I think we could just use the Faraday middleware in the API client, but that seems like a bigger change than should be included in this PR. Disability rating is not currently used on the front end, so I think you are safe to make any changes here to get specs passing and file an issue for us to revisit using a Faraday middleware.

**Comment by saneshark** [comment]
@robbiethegeek I want to make sure I am addressing you comment. lib/common/models/base.rb is in the vets-api repo, it has been for quite sometime, and multiple models inherit from it. Not sure what you mean by "is this still being worked and if so by whom." As for disability_ratings.rb, and disability_benefit_claim.rb I'm not entirely sure if it is being used, but I do believe that @carlmjohnson and @austin-martinez worked on it so tagged them in the PR. Aside from Rx and SM, those two models are the only ones that also inherit Common::Base.

I believe @austin-martinez will be addressing the broken specs in a commit to this branch shortly.

**Comment by austin-martinez** [comment]
@carlmjohnson Can you give a final review? I made some quick changes to the rating model/serializer to get the specs passing and created an issue for us to follow up on moving that conversion down to the Faraday middleware for the EVSS common services

**Pull request merged**